### ISSUE TITLE
[BUG] - Snap version no longer runs on old distros like Centos 7

### ISSUE THREAD

**Issue by danie-dejager** [aristocratos/btop#828]
**Read the README.md and search for similar issues before posting a bug report!**

Any bug that can be solved by just reading the [prerequisites](https://github.com/aristocratos/btop#prerequisites) section of the README will likely be ignored.

**Describe the bug**
I have installed btop 1.3.2 Rev 763 on Centos 7 x86_64 hosts. When I execute btop I get the following error:
` ERROR: Failed to create _runner thread!`

Snap version is current: 2.62
```
btop                    1.3.2     763    latest/stable  kz6fittycent  -
core22                  20240111  1122   latest/stable  canonical✓    base
snapd                   2.62      21465  latest/stable  canonical✓    snapd
```
[A clear and concise description of what the bug is.]

**To Reproduce**

Update btop to the latest version using snap.
Execute it

**Expected behavior**
Should still support older distros.

**Screenshots**
![image](https://github.com/aristocratos/btop/assets/13314239/a91a4c45-2e7e-4af2-a449-4a1f70df610a)

**Info (please complete the following information):**
```
name:      btop
summary:   Resource monitor that shows usage and stats
publisher: James Tigert (kz6fittycent)
store-url: https://snapcraft.io/btop
contact:   https://github.com/kz6fittycent/btop
license:   Apache-2.0
description: |
  Resource monitor that shows usage and stats for processor, memory, disks, network and processes.
  C++ version and continuation of bashtop and bpytop.
commands:
  - btop
snap-id:      UgRB8x3LtPA9L2Qo4G26ozMRImexxYzM
tracking:     latest/stable
refresh-date: today at 08:03 BST
channels:
  latest/stable:    1.3.2 2024-04-06 (763) 1MB -
  latest/candidate: ↑                          
  latest/beta:      ↑                          
  latest/edge:      1.3.2 2024-03-23 (763) 1MB -
installed:          1.3.2            (763) 1MB 

```
Kernel: 3.10.0-1160.108.1.el7.x86_64

**Additional context**

Contents of `~/.config/btop/btop.log`

Note: The snap uses: `~/snap/btop/current/.config/btop`
```
2024/04/24 (08:07:18) | ===> btop++ v.1.3.2
2024/04/24 (08:07:18) | DEBUG: Running in DEBUG mode!
2024/04/24 (08:07:18) | INFO: Logger set to DEBUG
2024/04/24 (08:07:18) | DEBUG: Using locale C.UTF-8
2024/04/24 (08:07:18) | WARNING: Error in Mem::collect() : Operation not permitted
2024/04/24 (08:07:18) | DEBUG: Shared::init() : Initialized.
2024/04/24 (08:07:18) | ERROR: Failed to create _runner thread!
2024/04/24 (08:07:18) | INFO: Quitting! Runtime: 00:00:00
```

(try running btop with `--debug` flag if btop.log is empty)

**GDB Backtrace**

If btop++ is crashing at start the following steps could be helpful:

(Extra helpful if compiled with `make OPTFLAGS="-O0 -g"`)

1. run (linux): `gdb btop` (macos): `lldb btop`

2. `r` to run, wait for crash and press enter if prompted, CTRL+L to clear screen if needed.
```
(gdb) r
Starting program: /var/lib/snapd/snap/bin/btop 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Missing separate debuginfo for /lib/libc.so.6
Try: yum --enablerepo='*debug*' install /usr/lib/debug/.build-id/94/70e279388f7f9cb2ed3b2872d0c2095b191ff4.debug
[New Thread 0x7fffd01bd700 (LWP 24921)]
[New Thread 0x7fffcf9bc700 (LWP 24922)]
[New Thread 0x7fffcf1bb700 (LWP 24924)]
[New Thread 0x7fffce9ba700 (LWP 24923)]
[New Thread 0x7fffce1b9700 (LWP 24925)]
[Detaching after vfork from child process 24926]
[New Thread 0x7fffcd06c700 (LWP 24932)]
[New Thread 0x7fffcc86b700 (LWP 24933)]
[Thread 0x7fffcc86b700 (LWP 24933) exited]
[Thread 0x7fffcd06c700 (LWP 24932) exited]
[Thread 0x7fffce1b9700 (LWP 24925) exited]
[Thread 0x7fffcf1bb700 (LWP 24924) exited]
[Thread 0x7fffcf9bc700 (LWP 24922) exited]
[Thread 0x7fffd01bd700 (LWP 24921) exited]
[Thread 0x7fffce9ba700 (LWP 24923) exited]
process 24917 is executing new program: /usr/libexec/snapd/snap-confine
Missing separate debuginfos, use: debuginfo-install glibc-2.17-326.el7_9.x86_64 openssl-libs-1.0.2k-26.el7_9.x86_64 zlib-1.2.7-21.el7_9.x86_64
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[Detaching after fork from child process 24934]
[Detaching after fork from child process 24935]
process 24917 is executing new program: /usr/lib/snapd/snap-exec
/usr/lib/snapd/snap-exec: No such file or directory.
Missing separate debuginfos, use: debuginfo-install snap-confine-2.62-0.el7.x86_64

```
3. (gdb): `thread apply all bt` (lldb): `bt all` to get backtrace for all threads
```
(gdb) bt all
No symbol table is loaded.  Use the "file" command.
(gdb) 

```
4. Copy and paste the backtrace here:
N/A

**Comment by shiyeling** [comment]
Same here~

**Comment by likecyber** [comment]
any update on this?

**Comment by paulgiordano** [comment]
I'm running into the same. CentOS release 7.8.2003. I know, I know, but I'm stuck with a couple of these machines and would really like btop on them. My traces and logs are as above.

**Comment by rescenic** [comment]
I have this issue as well.

**Comment by kz6fittycent** [comment]
I just happened to see this. Here's the problem, and I've had it before with Centos7. The RPM is broken:

Snaps use a small helper called `snap-exec` that launches the actual application with the correct confinement/profile.
On CentOS 7, the snapd package that ships snap-exec is broken/miscompiled or the file is simply missing from the RPM.
This is a very well-known, [long-standing issue](https://bugzilla.redhat.com/show_bug.cgi?id=1755646) with the official snapd package on CentOS 7 / RHEL 7. The binary `/usr/lib/snapd/snap-exec` (and sometimes `/usr/libexec/snapd/snap-confine`) is not included or is built for `glibc 2.18+` features that don’t exist on EL7, so it fails to appear on the filesystem → “No such file or directory”.

If you really want to use the snap, you can try this, but I don't believe it'll actually work:

```
# Completely remove the snap
sudo snap remove --purge btop

# Remove everything
sudo yum remove snapd
sudo rm -rf /snap /var/snap /var/lib/snapd

# Install the community-maintained fixed version (works on CentOS 7)
sudo yum install -y epel-release
sudo yum install -y snapd
sudo systemctl enable --now snapd.socket
sudo ln -s /var/lib/snapd/snap /snap

# Reinstall the snap
sudo snap install btop
```

Basically this is not the snap causing the issue, it's the `snapd` rpm, and it's something for those maintainer(s). I don't know if they're going to help with it at this point, but this is the issue you're facing. 

@aristocratos - this issue can be closed. The users should install the binary as supplied by this repo or compile it locally instead of using the snap.
